crates/apollo-router-core/src/federated.rs

```diff
@@ -118,42 +118,47 @@ impl Fetcher for FederatedGraph {
 
-                tracing::debug!("query plan\n{:#?}", &plan);
+                if let Some(plan) = query_plan.node() {
+                    tracing::debug!("query plan\n{:#?}", plan);
 
-                let early_errors_response = tracing::info_span!("validation").in_scope(|| {
-                    let mut early_errors = Vec::new();
-                    for err in validate_services_against_plan(Arc::clone(&service_registry), &plan)
-                    {
-                        early_errors.push(err.to_graphql_error(None));
-                    }
+                    let early_errors_response = tracing::info_span!("validation").in_scope(|| {
+                        let mut early_errors = Vec::new();
+                        for err in
+                            validate_services_against_plan(Arc::clone(&service_registry), plan)
+                        {
+                            early_errors.push(err.to_graphql_error(None));
+                        }
 
-                    for err in validate_request_variables_against_plan(Arc::clone(&request), &plan)
-                    {
-                        early_errors.push(err.to_graphql_error(None));
-                    }
+                        for err in
+                            validate_request_variables_against_plan(Arc::clone(&request), plan)
+                        {
+                            early_errors.push(err.to_graphql_error(None));
+                        }
 
-                    // If we have any errors so far then let's abort the query
-                    // Planning/validation/variables are candidates to abort.
-                    if !early_errors.is_empty() {
-                        tracing::error!(errors = format!("{:?}", early_errors).as_str());
-                        let response = Response::builder().errors(early_errors).build();
-                        Some(stream::once(async move { response }).boxed())
-                    } else {
-                        None
-                    }
-                });
+                        // If we have any errors so far then let's abort the query
+                        // Planning/validation/variables are candidates to abort.
+                        if !early_errors.is_empty() {
+                            tracing::error!(errors = format!("{:?}", early_errors).as_str());
+                            let response = Response::builder().errors(early_errors).build();
+                            Some(stream::once(async move { response }).boxed())
+                        } else {
+                            None
+                        }
+                    });
 
-                if let Some(response) = early_errors_response {
-                    return response;
+                    if let Some(response) = early_errors_response {
+                        return response;
+                    }
+                } else {
+                    return stream::empty().boxed();
                 }
```

Comment:
I wonder in what context this can happen ?

Reply:
I have no idea to be honest. Apparently the planner can return a json object without the key "node" at all. Maybe @BrynCooke knows? Otherwise it would require checking the planner's code

Reply:
It seems that you can request:
```
query {__typename}
```
In theory I guess this doesn't actually make any backend calls, and the query plan in it's current form can't represent this query?

Probably a bug but way down the list for us to fix.

Reply:
Yes most likely an uninteresting bug that will solve itself when we riir the query planner.

Reply:
a bug needs an issue though :wink: